### PR TITLE
re-introduce updated CSCS CI

### PR DESCRIPTION
**Pull request by boeschf**
unit tests and examples now run again on CSCS premises:
- eiger: 2x AMD EPYC 7742 (multicore)
- todi: 4x GH200 modules  (gpu: arbor is compiled with cuda 12.4 and runs on the H100)

The distributed tests are not enabled as of now (due to current problems with the gitlab runners). Once this is fixed, I will add another PR to enable them.

The tests run automatically for every PR to master, but only for whitelisted users. Manual trigger: make a comment with the content `cscs-ci run default`.

Other changes:
- use rpaths for pyarb: helps the linker to find pugixml
- mark some more external CMake variables as advanced (from units)

### CODE REVIEW

**Comment by boeschf** [comment]
cscs-ci run default

**Comment by thorstenhater** [comment]
Don't worry too much about Spack, something is wrong with their infrastructure...
```
 ==> Installing gettext-0.22.5-n34hduanct2b5qie3ri23koih2kgo7ty [25/31]
==> No binary for gettext-0.22.5-n34hduanct2b5qie3ri23koih2kgo7ty found: installing from source
==> Warning: The contents of /tmp/runner/spack-stage/spack-stage-gettext-0.22.5-n34hduanct2b5qie3ri23koih2kgo7ty/gettext-0.22.5.tar.xz look like HTML.  Either the URL you are trying to use does not exist or you have an internet gateway issue.  You can remove the bad archive using 'spack clean <package>', then try again using the correct URL.
```

**Comment by boeschf** [comment]
cscs-ci run

**Comment by boeschf** [comment]
cscs-ci run default

**Comment by boeschf** [comment]
cscs-ci run default

File: CMakeLists.txt [hunk 1]
```diff
@@ -323,6 +323,7 @@ if (ARB_WITH_PYTHON)
 endif()
 
 CPMAddPackage(NAME pugixml
+              FORCE TRUE
```
Comment (thorstenhater) [inline CMakeLists.txt:326]: Why is that needed? Could pugi be provided by the OS/base layer?

File: CMakeLists.txt [hunk 2]
```diff
@@ -392,7 +392,7 @@ mark_as_advanced(FORCE RANDOM123_INCLUDE_DIR)
 mark_as_advanced(FORCE pybind11_DIR PYBIND11_PYTHONLIBS_OVERWRITE PYBIND11_PYTHON_VERSION PYBIND11_FINDPYTHON PYBIND11_INSTALL PYBIND11_INTERNALS_VERSION PYBIND11_NOPYTHON PYBIND11_SIMPLE_GIL_MANAGEMENT PYBIND11_TEST)
 mark_as_advanced(FORCE pugixml_DIR)
 mark_as_advanced(FORCE fmt_DIR)
-mark_as_advanced(FORCE units_DIR UNITS_BUILD_OBJECT_LIBRARY UNITS_BUILD_SHARED_LIBRARY UNITS_HEADER_ONLY UNITS_NAMESPACE)
+mark_as_advanced(FORCE units_DIR UNITS_BUILD_OBJECT_LIBRARY UNITS_BUILD_SHARED_LIBRARY UNITS_HEADER_ONLY UNITS_NAMESPACE UNITS_BUILD_FUZZ_TARGETS UNITS_ENABLE_TESTS)
```
Comment (thorstenhater) [inline CMakeLists.txt:395]: Yes.

**Review by thorstenhater — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (boeschf) [reply CMakeLists.txt]: I reintroduced the system package, if available. Changed the install method for pyarb to use rpaths, however, for it to find pugixml.so when linked with a spack installed version, for example.

File: python/CMakeLists.txt [hunk 3]
```diff
@@ -49,10 +49,20 @@ set_target_properties(pyarb PROPERTIES OUTPUT_NAME _arbor)
 #   arbor.cpython-36m-x86_64-linux-gnu.so
 set_target_properties(pyarb PROPERTIES PREFIX "${PYTHON_MODULE_PREFIX}" SUFFIX "${PYTHON_MODULE_EXTENSION}")
 
+# Set RPATH for the installation phase
+if(APPLE)
```
Comment (thorstenhater) [inline python/CMakeLists.txt:53]: I am fine with that solution, although I was originally thinking about using
it in header-only mode still. Nothing that can't be change later, though, if
needed.

**Review by thorstenhater — APPROVED** [review]
Thanks 🎉
